imcsdk/apis/admin/ip.py

```diff
@@ -0,0 +1,291 @@
+                            Range [300-900] seconds
+        kwargs: Key-Value paired arguments for future use
+
+    Returns:
+        IpBlocking object
+    """
+
+    mo = IpBlocking(parent_mo_or_dn=_get_mgmt_if_dn(handle))
+
+    params = {
+        'enable': 'yes',
+        'fail_count': fail_count,
+        'fail_window': fail_window,
+        'penalty_time': penalty_time
+    }
+    mo.set_prop_multiple(**params)
+    mo.set_prop_multiple(**kwargs)
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def ip_blocking_disable(handle):
+    """
+    Disables IP Blocking
+
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        None
+    """
+    mo = IpBlocking(parent_mo_or_dn=_get_mgmt_if_dn(handle))
+    mo.enable = 'no'
+    handle.set_mo(mo)
+
+
+def is_ip_blocking_enabled(handle,):
+    """
+    Checks if IP Blocking is enabled
+
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        bool
+    """
+    mo = IpBlocking(parent_mo_or_dn=_get_mgmt_if_dn(handle))
+    mo = handle.query_dn(mo.dn)
+    return mo.enable.lower() in ['yes', 'true']
+
+
+def ip_blocking_exists(handle, **kwargs):
+    """
+    Checks if IP blocking settings match according to the parameters specified.
+
+    Args:
+        handle (ImcHandle)
+        kwargs: Key-Value paired arguments relevant to IpBlocking object
+
+    Returns:
+        (True, IpBlocking object) if exists, else (False, None)
+    """
+    mo = IpBlocking(parent_mo_or_dn=_get_mgmt_if_dn(handle))
+    mo = _get_mo(handle, dn=mo.dn)
+
+    params = {
+        'enable': kwargs.get('enable'),
+        'fail_count': kwargs.get('fail_count'),
+        'fail_window': kwargs.get('fail_window'),
+        'penalty_time': kwargs.get('penalty_time')
+    }
+
+    if mo.check_prop_match(**params):
+        return (True, mo)
+
+    return (False, None)
+
+
+_IP_FILTER_LIST = ['filter1', 'filter2', 'filter3', 'filter4']
+
+
+def _set_ip_filters(mo, filters):
+    if len(filters) > len(_IP_FILTER_LIST):
+        raise ImcOperationError("Set Ip Filters",
+                                "Cannot specify more than %d filters" %
+                                len(_IP_FILTER_LIST))
+
+    params = {'filter' + str(x['id']): str(x['filter']) for x in filters}
+    mo.set_prop_multiple(**params)
+
+
+def ip_filtering_enable(handle, filters=[]):
+    """
+    Enables IP filtering with the filters specified
+    Connection is expected to drop.
+
+    Args:
+        handle (ImcHandle)
+        filters (list): List of dictionaries of the format
+            {'id': 1, 'filter': "10.10.10.10",
+             'id': 2, 'filter': "2.2.2.2-3.3.3.3"}
+
+    Returns:
+        IpFiltering object
+    """
+    mo = IpFiltering(parent_mo_or_dn=_get_mgmt_if_dn(handle))
+    mo.enable = 'yes'
+    _set_ip_filters(mo, filters)
+
+    handle.set_mo(mo)
+    # Not returning the server copy as the connection drops after the above action
+    return mo
+
```

Comment:
Good to add warning message as this will logout the existing user and not allow any other IP to login. Also applicable for modify case.